include/swift/Sema/IDETypeChecking.h

```diff
@@ -50,6 +51,8 @@ namespace swift {
   class ConstraintSystem;
   class Solution;
   class SolutionApplicationTarget;
+  enum class ConstraintSystemFlags;
```

Comment:
While this makes me slightly uncomfortable, if there is ever a size mismatch between the forward declaration and the full definition with enum members, there is a compile-time error.

e.g.

```
swift/include/swift/Sema/IDETypeChecking.h:54:14: error: enumeration redeclared with different underlying type 'long' (was 'int')
  enum class ConstraintSystemFlags : long;
             ^
swift/include/swift/Sema/ConstraintSystem.h:1453:12: note: previous declaration is here
enum class ConstraintSystemFlags {
           ^
```